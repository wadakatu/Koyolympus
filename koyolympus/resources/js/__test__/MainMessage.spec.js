import {shallowMount} from '@vue/test-utils';
import Component from '@/MainMessageComponent.vue';

describe('Testing methods', () => {
    test('Testing photo method', () => {
        const wrapper = shallowMount(Component, {
            mocks: {
                $store: {commit: jest.fn()},
                $router: {
                    push: jest.fn().mockResolvedValue({})
                }
            }
        });
        wrapper.vm.photo();

        expect(wrapper.vm.$store.commit).toHaveBeenCalledTimes(2);
        expect(wrapper.vm.$store.commit).toHaveBeenCalledWith('photo/setUrl', '/photo/random');
        expect(wrapper.vm.$store.commit).toHaveBeenCalledWith('photo/setGenre', null);
        expect(wrapper.vm.$router.push).toHaveBeenCalled();
        expect(wrapper.vm.$router.push).toHaveBeenCalledWith({name: 'photo.random'});
    });
});

describe('Testing @ event', () => {
    test('called photo method when click explore button', async () => {
        const spy = jest.spyOn(Component.methods, 'photo');
        const wrapper = shallowMount(Component, {
            mocks: {
                $store: {commit: jest.fn()},
                $router: {
                    push: jest.fn().mockResolvedValue({})
                }
            }
        });

        wrapper.find('.photo_button').trigger('click');
        await wrapper.vm.$nextTick();
        expect(spy).toHaveBeenCalled();
    });
});

describe('Testing SnapShot', () => {
    test('Default', () => {
        const wrapper = shallowMount(Component, {
            mocks: {
                $store: {commit: jest.fn()},
                $router: {
                    push: jest.fn().mockResolvedValue({})
                }
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});
