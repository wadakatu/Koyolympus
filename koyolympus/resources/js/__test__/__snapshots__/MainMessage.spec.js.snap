// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SnapShot Default 1`] = `
<div
  class="col"
>
  <div
    class="profile"
  >
    <h1
      class="title"
    >
      Koyolympus
    </h1>
     
    <p
      class="description"
    >
      
            Photography is a means of capturing and preserving the instant that doesn't come twice.
            That's why I cannot stop exploring the world with my camera gear to cut out the exact moment
            that I would like to keep as it is.
        
    </p>
     
    <button
      class="photo_button"
    >
      Explore My Photo
    </button>
  </div>
</div>
`;
